hvplot/plotting/scatter_matrix.py

```diff
@@ -38,19 +49,74 @@ def scatter_matrix(data, c=None, chart='scatter', diagonal='hist', alpha=0.5, **
     diagonal = HoloViewsConverter._kind_mapping[diagonal]
     chart = HoloViewsConverter._kind_mapping[chart]
 
+    #remove datashade kwds
+    if datashade == True or rasterize == True:
+        ds_kwds = {}
+        if 'aggregator' in kwds:
+            ds_kwds['aggregator'] = kwds.pop('aggregator')
+        
+    #remove dynspread kwds
+    if dynspread:
+        dn_kwds = {}
+        if 'max_px' in kwds:
+            dn_kwds['max_px'] = kwds.pop('max_px')
+        if 'threshold' in kwds:
+            dn_kwds['threshold'] = kwds.pop('threshold')
+        if 'how' in kwds:
+            dn_kwds['how'] = kwds.pop('how')
+        if 'mask' in kwds:
+            dn_kwds['mask'] = kwds.pop('mask')
+
+    if spread:
+        sp_kwds = {}
+        if 'px' in kwds:
+            sp_kwds['px'] = kwds.pop('px')
+        if 'shape' in kwds:
+            sp_kwds['shape'] = kwds.pop('shape')
+        if 'how' in kwds:
+            sp_kwds['how'] = kwds.pop('how')
+        if 'mask' in kwds:
+            sp_kwds['mask'] = kwds.pop('mask')
+        
+
     colors = _hv.plotting.util.process_cmap('Category10', categorical=True)
     chart_opts = dict(alpha=alpha, cmap=colors, tools=['box_select', 'lasso_select'],
                       nonselection_alpha=0.1, **kwds)
 
     grid = _hv.operation.gridmatrix(data, diagonal_type=diagonal, chart_type=chart)
+
     if c:
         chart_opts['color_index'] = c
         grid = grid.map(lambda x: x.clone(vdims=x.vdims+[c]), 'Scatter')
         groups = _hv.operation.gridmatrix(data.groupby(c).overlay(),
                                           chart_type=chart,
                                           diagonal_type=diagonal)
-        grid = (grid * groups).map(lambda x: x.get(0) if isinstance(x.get(0), chart) else x.get(1),
-                                   _hv.Overlay)
+        grid = grid * groups
```

Comment:
I'm having trouble following the logic here, but in the old code `.map` was applied only under the `if c:` branch, but in the new code the indentation indicates that it's applied regardless of `c`.  Is that a problem?

Reply:
The new changes look good, thanks!  I still don't see the answer to this question; it seems like the code now in `apply_to_charts` was previously run only when `c` was non-None, and now it's always run. Is that intentional?  How was it working before if so? @philippjfr ?

Reply:
I'm still working on this.  I think Jean-Luc agreed with you about moving the .map part this back under the c requirement.  I think I moved it, because it wouldn't apply the datashade/rasterize/etc if you didn't have c specified.  C was giving it a column to color code by.  Can you datashade without this color coding requirement?  If not, then the previous way was correct.  
anyway, this is not working correctly anyway with the fixes.  Right now, it seems to do something with datashade, but when I try to rasterize or give it a new value for dynspread, for example, it seems to ignore it.  So, I'm trying to figure out where I had it working and what I did wrong in the process of making corrections.

Reply:
My guess is that because of this change, something either won't work when `c` is specified either in the old code or the new code. I can't quite tell what the old code was meant to do, but hopefully @philippjfr can.

Reply:
I made some comments about what each step is doing in this `if c` block.  All these lines are necessary to get the groups to color properly within each element in the scatter matrix.  I have moved things so that the datashade functions are performed after all the coloring is complete.